### DISCUSSION TITLE
[hast] - "@types/hast": "3.0.4"

### DISCUSSION THREAD

**Discussion by babakfp** [DefinitelyTyped/DefinitelyTyped#69775] [Issues with a @types package]
```
[21:48:13] Starting compilation in watch mode...

node_modules/.pnpm/@types+hast@3.0.4/node_modules/@types/hast/index.d.ts:205:18 - error TS2430: Interface 'Element' incorrectly extends interface 'Parent'.
  Types of property 'data' are incompatible.
    Type 'ElementData | undefined' is not assignable to type 'Data | undefined'.
      Type 'ElementData' is not assignable to type 'Data'.
        Types of property 'meta' are incompatible.
          Type 'string | null | undefined' is not assignable to type 'string | undefined'.
            Type 'null' is not assignable to type 'string | undefined'.

205 export interface Element extends Parent {
                     ~~~~~~~

node_modules/.pnpm/@types+hast@3.0.4/node_modules/@types/hast/index.d.ts:236:18 - error TS2430: Interface 'ElementData' incorrectly extends interface 'Data'.

236 export interface ElementData extends Data {}
                     ~~~~~~~~~~~

[21:48:17] Found 2 errors. Watching for file changes.
```

> [!NOTE]
> It works fine on version `3.0.3`.

[a comment by an automated account is not shown]

**Comment by ChristianMurphy** [comment]
@babakfp can you share your TSConfig and the version of `@types/unist` you are using?
I tried reproducing with `@types/hast` version 3.0.4 and `@types/unist` version 3.0.2, with `typescript` version 5.4.5 along with a strict configuration:

```json
{
  "compilerOptions": {
    "target": "ES2020",
    "useDefineForClassFields": true,
    "lib": ["ES2020", "DOM", "DOM.Iterable"],
    "module": "Node16",
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noFallthroughCasesInSwitch": true
  },
  "include": ["src"],
  "references": [{ "path": "./tsconfig.node.json" }]
}
```

and see no error.

reproducible example: https://stackblitz.com/edit/github-wvmqw9?file=package-lock.json

**Reply by babakfp** [reply]
Hi 👋

This is the project: https://github.com/babakfp/svelte-in-markdown
Change the version number to `3.0.4` and do `pnpm i && pnpm dev`.

TSConfig:

```json
{
    "compilerOptions": {
        "module": "NodeNext",
        "moduleResolution": "NodeNext",
        "outDir": "dist",
        "strict": true,
        "declaration": true
    },
    "include": ["./src/**/*"]
}
```

There is no `@types/unist` dep (in package.json).

**Reply by ChristianMurphy** [reply]
@babakfp can you create a minimal reproducible example? (https://stackoverflow.com/help/minimal-reproducible-example)
Your project has over nine plugins and multiple module overrides.

https://stackblitz.com/github/babakfp/svelte-in-markdown could be a good starting point

**Reply by ChristianMurphy** [reply]
For example, with your exact configuration when I clear all the other plugins and module overrides, and keep just the Element import, the types resolve as expected with your TypeScript configuration.
https://stackblitz.com/edit/github-2qtpyh?file=src%2Findex.ts

I strongly suspect this is a broken override in your code rather than a broken upstream typing.

**Reply by babakfp** [reply]
I stripped down the whole project to only `unified` `remark-parse` `remark-rehype` **`@shikijs/rehype`** `rehype-stringify`.

The type issue is still there. I think this may be because of the shiki plugin. When I remove shiki, the error goes away. There is a `hast` import in shiki plugin.

https://github.com/babakfp/svelte-in-markdown/tree/hast_ts_issue_3

**Reply by ChristianMurphy** [reply]
Thanks for helping to isolate it @babakfp!
I suspect the issue is this snippet in shiki https://github.com/shikijs/shiki/blob/2469296ccebad1eef8d4a14716ec3aa55591372c/packages/rehype/src/core.ts#L61-L68

/cc @remcohaszing is there a specific order the module maps need to be entered to ensure they override correctly?

**Reply by remcohaszing** [reply]
Not that I know of.

Anyway, see https://github.com/shikijs/shiki/pull/699
